app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsFragment.kt

```diff
@@ -321,6 +328,7 @@ class TransactionDetailsFragment : BaseViewBindingFragment<FragmentTransactionDe
 
         binding.content.visible(true)
         binding.contentNoData.root.visible(false)
```

Comment:
better hanlde loading visibility in viewModel.state.observe (UpdateDetails viewAction in this case) and let updateUi deal only with content elements of the page

Reply:
If you are loading content into the page, shouldn't the loading spinner always be hidden when the content is received? Besides, remember we had issues with `liveData` swallowing some state changes if you emit too frequently? I think this is a bit safer and makes sense in terms of how we use the app components. As in,

if loading -> show loading
if error -> hide content, hide loading, show error
if content -> hide loading, hide error, show content

Reply:
we only call showContent in UpdateDetails

Reply:
yes, what you said is right, but having it separated and being called explicitly makes code more readable I think

Reply:
even in your description you have separated those steps (hide loading, show content)

Reply:
yeah but what I am talking about is the liveData that swallows events. That's what I think is a bigger problem.

Reply:
to be clear: I am asking to sepparately call showLoading(false) in UpdateDetails and remove code that deails with spinners from updateUi function.